### ISSUE TITLE
Plasma Desktop - Issues

### ISSUE THREAD

**Issue by rbharathkumar** [clearlinux/distribution#381]
**Describe the bug**
As per the suggestion from @anselmolsm , I decided to retry KDE on a VM once again.
The following are some of the observations. I will let more learned people decide if they are bugs.
1. After the installation of the desktop-kde and restarting the system does not boot into the login manager. SDDM is in inactive status. But unlike in XFCE, enabling and starting boots to the Login manager.
2. The resolution of the Login Manager is not proper. It gives a small window in a black screen. Even after restarts.
3. There is no indication that the focus is available on the password menu. A blinking cursor would be helpful. Since the first login always fails. Then we need to type in the password and login again.
4. The login manager nor the desktop has Shutdown or Restart buttons available. An administrator user should be having those options.
5. On login, the default font of the KDE desktop is very light. Almost impossible to read. Easily fixable. Not a bug more of a gripe.

**To Reproduce**
1. Perform the default install without installing desktop bundle.
2. Reboot.
3. Update if available.
4. Install desktop-kde.

**Comment by anselmolsm** [comment]
Thanks for the test and report @rbharathkumar . Some thoughts:

* 1 desktop-kde has do what desktop-autostart does (perhaps a bundle desktop-kde-autostart) to deal with systemd as necessary;
* 2, 3 and 4 may be fixed as part of changes to address #380 
* 5 It might be due to desktop-kde bundle alone not installing fonts enough. That's easy to fix.

We'll keep this issue updated as changes get integrated.

**Comment by rbharathkumar** [comment]
Thank you for your response @anselmolsm 
Really good to know that fixes are already in progress. Would love to test and help out.

Regarding point #5, by default we only have a three fonts available.
I had installed the Noto Fonts, updated the cache (After installing devpkg-fontconfig) and modified the systems fonts to use the Noto Fonts, but the rendering still seems off. It is very light and almost invisible to see in the default Breeze theme. Breeze-Dark makes it bearable. but just so.

Few other observations,
6. The default install comes up with way too many applications. The entire KDE meta-apps, this can be trimmed to provide the necessary ones. 
7. Despite the whole list of applications, somehow a proper browser application and software center is missing. Konqueror is available. but Falcon would have been better. Similarly would be really easy to have Discover installed similar to how the Gnome Software Center is available.

Thank you once again for all the great work. Appreciate it.

**Comment by fenrus75** [comment]
https://github.com/clearlinux/clr-bundles/blob/master/bundles/desktop-kde
https://github.com/clearlinux/clr-bundles/blob/master/bundles/desktop-kde-apps

is the source of what describes what goes into the bundles...

makes it easier to be specific ;)

**Comment by rbharathkumar** [comment]
My suggestion would be to remove the desktop-kde-apps bundle from the desktop-kde bundle.
If the user needs the extra kde apps, they can go ahead and install them separately.

But this would mean that the users installing only the desktop-kde may not have a usable system. 
So we can include a few basic apps in the desktop-kde bundle. 
One of Each instead of the whole KDE application set.


Archive Manager - Ark
File Manager - Dolphin + Plugins
Web Browser - Falcon/Firefox
Notes Application - Kate/Atom
Screenshot tool - Spectacle
Image Viewer - Gwenview
Document Viewer - Okular
Terminal - Konsole
Task Manager - KSysLog
Audio/Video Player - VLC
Software Center - Discover

Optional: Some can be installed via Flatpaks
Calendar App
Mail App - Thunderbird/KMail
Printer/Scan/Bluetooth Support - Skanlite

We can completely do away with the games from the basic desktop-kde apps.

**Comment by fenrus75** [comment]
this will be a few steps so that existing installs don't suddenly lose apps
the user liked

step 0: stop desktop-kde from including full desktop-kde-apps (done now)
step 1: add separate games bundle, still included from desktop-kde-apps
step 2: (later) stop the include

existing users will get the games bundle installed in step 1
and after step 2 it becomes optional

**Comment by rbharathkumar** [comment]
Thank you for your response and for including Discover in the desktop-kde bundle.
Yes, it looks like a good balance, although I would really suggest including basic applications to the desktop-kde bundle. 
Please consider one of each. If not possible atleast Dolphin and Konsole. A desktop definitely needs its file manager and terminal applications. Others can be installed via other bundles or flatpaks. Otherwise the user is again forced to install the desktop-kde-apps bundle and the advantage of splitting the apps goes away.

My apologies if they already come bundled as part of something other packages.

**Comment by anselmolsm** [comment]
As of 27550: Fixes & changes addressing items 1 to 5 released.
plasma-discover packaged and included to desktop-kde bundle (there was another discover there before).

**Comment by rbharathkumar** [comment]
Thank you so much @anselmolsm 
I will create a new VM and test out the new build.

I am really looking forward to the font management fix more than anything. It was really hard to work on the desktop with illegible lettering. Will test and let you know. thank you once again.

**Comment by rbharathkumar** [comment]
Hi @anselmolsm ,
As discussed, tried out the new build in a VM. Installed from scratch.
Please find the bundle list attached to this post.

Let me start with the things that have changed positively.
1. First and foremost, the difference in font rendering is night and day. It looks amazing. Just like how KDE looks in my local system. So cheers for that.
2. The software list has been rebuilt. I cannot believe that the DEVs took suggestion from a end-user. I hope this kind of attitude towards user-feedback continues.
3. The software stack is now very minimal and just like how I had expected. But one suggestion is that we can include a few additional tools alone. (Archive Manager, one GUI based text editor and Screenshot tool). Since flatpak refuses to work, I had to install atom and firefox from the bundles to even write this from within the VM.
4. The login screen is now better with focus on the password menu. Auto Start works just as expected.

Now onto the things that do not work.
1. SDDM still does scale to full resolution. Even after the restarts. But when I lock the system, it scales and renders in full screen.
2. The application menu still does not have a Shutdown/Retart button. SDDM does have it. This is for an administrator user created during the install process.
3. Flatpak just refuse to work. Requests for TLS support. Is there any other package I need to install to use ? This must be pre-installed for the flathub website clearly states that in Clear Linux flatpaks come pre-configured upon installing the flatpak bundle.

```
rbkclear@clearlinuxkde ~ $ flatpak remote-add flathub https://flathub.org/repo/flathub.flatpakrepo
Can't load uri https://flathub.org/repo/flathub.flatpakrepo: TLS support is not available
rbkclear@clearlinuxkde ~ $ 

```
4. Thank you for including the Discover app, but it does not start. The same case with System Settings. In fact system settings gives out a Crashed Message asking for an application restart. I had restarted several times but to no avail.

5. I tried invoking Discover from the Konsole,it gives the following error message.

```


rbkclear@clearlinuxkde ~ $ plasma-discover %F
QQmlApplicationEngine failed to load component
qrc:/qml/DiscoverWindow.qml:6 module "org.kde.kirigami" is not installed

org.kde.plasma.discover: Errors when loading the GUI
org.kde.plasma.discover: Trying to open unexisting file QUrl("file:///home/rbkclear/%25F")
invalid kns backend! "/usr/share/xdg/servicemenu.knsrc" because: "Config group not found! Check your KNS3 installation."
invalid kns backend! "/usr/share/xdg/ksysguard.knsrc" because: "Config group not found! Check your KNS3 installation."
org.kde.plasma.libdiscover: couldn't find category decoration "pkcategories/applicationcrop.jpg"
org.kde.plasma.libdiscover: couldn't find category decoration "pkcategories/accessoriescrop.jpg"
org.kde.plasma.libdiscover: couldn't find category decoration "pkcategories/accesscropped.jpg"
org.kde.plasma.libdiscover: couldn't find category decoration "pkcategories/toolcrop.jpg"
org.kde.plasma.libdiscover: couldn't find category decoration "pkcategories/educationcrop.jpg"
org.kde.plasma.libdiscover: couldn't find category decoration "pkcategories/sciencecrop.jpg"
org.kde.plasma.libdiscover: couldn't find category decoration "pkcategories/gamecrop.jpg"
org.kde.plasma.libdiscover: couldn't find category decoration "pkcategories/graphiccrop.jpg"
org.kde.plasma.libdiscover: couldn't find category decoration "pkcategories/internetcrop.jpg"
org.kde.plasma.libdiscover: couldn't find category decoration "pkcategories/multimediacrop.jpg"
org.kde.plasma.libdiscover: couldn't find category decoration "pkcategories/officecrop.jpg"
org.kde.plasma.libdiscover: couldn't find category decoration "pkcategories/settingscrop.jpg"

```

Other App related Issues:
1. I had to use Atom to type out this long message. But when I tried to save or open a file through Atom it would crash out.

2. Firefox gives out a ton of error message about some pixbuf issues. On its default state when the title bar is removed, the buttons are not visible. The buttons are there, but not visible.

3. I still have not tested the audio issues I faced earlier.

Will let you know if I come across any further issues.
[bundle_list.txt](https://github.com/clearlinux/distribution/files/2822135/bundle_list.txt)

**Comment by anselmolsm** [comment]
Hi @rbharathkumar, thank you very much for the feedback and also for all the details provided. I'll use your bundle_list to reproduce your environment to find additional fixes needed.

**Comment by anselmolsm** [comment]
Quick update: fixes for audio/suspend/shutdown/reboot buttons, systemsettings5 & plasma-discover, and flatpak TLS issue are on their way to get released in the next couple of days.

**Comment by rbharathkumar** [comment]
Hi @anselmolsm , thank you for the update.
I will be creating a new VM and performing a complete install to get started once it is ready.
1. Based on the bundle_list I had provided, can you suggest the list of bundles to get started, I usually install desktop-kde, sysadmin-basic, network-basic and flatpak bundles. This is apart from the usual default. Is there anything else to be installed ? Maybe alsa-utils ?
2. Will the update also fix the other software related issues we faced in firefox (pixbuf & gstreamer) library issues and the "Save Option" in Atom ? Similarly with the SDDM resolutions ?
3. Any chances to include a screenshot tool, GUI base text editor and archive manager with the default desktop-kde install ?

**Comment by anselmolsm** [comment]
Hi @rbharathkumar , v27660 released today (Feb 6) includes fixes to the issues I mentioned in my previous comment, except the flatpak pkg update that may be available within 1 or 2 days.

* 1 & 3 - Hard to tell, it depends on what you need... Have you seen [this page](https://clearlinux.org/documentation/clear-linux/reference/bundles#bundles) with the list of bundles with descriptions? 
[desktop-kde-apps](https://github.com/clearlinux/clr-bundles/blob/master/bundles/desktop-kde-apps) contains the applications you mentioned in 3, perhaps some of them could move to desktop-kde (still not convinced about moving all)... dolphin is already there, maybe kwrite if kate package gets split, not sure.

* 2 - sddm resolution I could not reproduce on real hw yet, will try on a VM. I strongly recommend separate reports about Firefox and Atom - it'll also make it easier to keep track of what is being addressed here as well as in the separate(s) issue(s) ;)

**Comment by rbharathkumar** [comment]
> the flatpak pkg update that may be available within 1 or 2 days.

Thank you so much for the update. I will wait for the flatpak pkg update to complete before I start the install and thank you for your efforts in resolving all the issues.

1 & 3 
Yes I have been through the bundle list several times. Let me explain my reasoning behind the query. For the gnome desktop all I have to install is the desktop-autostart bundle. It basically brings in all the pre-requisites or atleast I think it does.

When I install the desktop-kde, I am confused if I should install any other bundles...alsa-utils is an example and when the sound did not work it made me wonder if installing the bundle would solve it. (Just an example, of course the issue was different). It was the same case with firefox with the pixbuff issues, so was there any other bundles I needed to install to get it working ? Of course the issue stems from technical ignorance from my end, but still confusion could have been avoided.

So I am able to think of two possible suggestions.
1. We can include the pre-requisite bundles along with the desktop-kde
2. Have some documentation that for the desktop-kde to work properly these bundles are also needed.

Of course I am not the expert, so the whole point may be moot since the dependencies may already be available as it is.

Regarding the apps, I am completely against moving all the desktop-kde-apps to the desktop-kde bundle. KDE ships a boatload of applications. I prefer your approach. Considering that Dolphin ,Discover and Konsole are already there, would be nice to just include kwrite or kate, ark and spectacle. Others can be added via flatpaks. 

Maybe we can have KDE's flatpak repository pre-configured or atleast made available as an option in Discover. 

Once again I appreciate all the work you are putting in. Please let me know if I can help in the development or testing in anyway.

**Comment by anselmolsm** [comment]
Hi @rbharathkumar, 

I see. I think it's more about the lack of something like a desktop-kde-autostart analog to what desktop-autostart is. Note taken.

**Comment by rbharathkumar** [comment]
Yes, more or less that is the point.
As an extension what I am suggesting is educating the end user on the dependencies or having them as you suggested already available.

I also strongly suggest that we have similar options for the xfce-desktop as well, 

What you have done for the KDE desktop by having the SDDM pre-configured is truly useful. The xfce-desktop stack could use a similar effort.

**Comment by rbharathkumar** [comment]
One more observation in the KDE desktop.
I am unable to install the guest additions. The clear-linux website says
`
The kernel modules are shipped with the kernel-lts bundle. Insert Guest Additions CD image using Devices menu you’ll need to install the user Linux Guest Additions. To install the VirtualBox Guest Additions, follow these steps:`

I have added the guest additions iso image via the devices option, but the kernel does not seem to be having those modules. I am getting the below error message.```

$ sudo install-vbox-lga
Password: 
ERROR: VBOX KERNEL MODULES NOT FOUND
Please use

        swupd verify --fix

        to fix your installation.
```

**Comment by anselmolsm** [comment]
Hi @rbharathkumar, please report the vbox problem in a separate issue (that's really not related to desktop-kde bundle).

BTW, flatpak TKS fix is now available.

**Comment by rbharathkumar** [comment]
Thank you, I will download the installer and perform a new install and let you know of the issues.
I will also open a separate issue for the vbox problem.

**Comment by rbharathkumar** [comment]
Just installed the latest version. Yet to test the various features.
Wanted to give a quick idea on the flatpak option alone. Still has some issues.
```

rbkclear@clearlinuxkde ~ $ flatpak remote-add --if-not-exists flathub https://flathub.org/repo/flathub.flatpakrepo

(flatpak remote-add:1469): GLib-GIO-ERROR **: 08:32:31.560: No GSettings schemas are installed on the system
Trace/breakpoint trap (core dumped)

```

Will let you know about the other things once testing is complete.

**Comment by Riotvan81** [comment]
Hi, could you please make a bundle for the kde dev libs? I'm trying to compile latte-dock but it's complaining that it can't find KF5Activities. While libKF5Activities is present libKF5Activities-dev is not. Or maybe just add latte-dock somewhere since it's quite popular among kde users.

**Comment by fenrus75** [comment]
I've added both
the dev bundle will take another workday since those get processed daily

the dock app ought to go out today still

**Comment by Riotvan81** [comment]
Thanks allot!

**Comment by rbharathkumar** [comment]
Hi @anselmolsm any news on the flatpak issue ? 
Would love to test how discover displays the flatpak apps not just from flathub but from other sources as well...

Also would be nice to know if any decision was reached on including kate, spectacle and ark (not the entire KDE app stack, just those three since dolphin, konsole and discover are already included) with the default desktop-kde bundle

**Comment by anselmolsm** [comment]
Hi @rbharathkumar , The flatpak update dealing with TLS issue has been published (I did not post update about that back then, I apologize.). I'll double check the glib error you posted (I does not happen on my system, but I'll check on a fresh env).

Ongoing changes in bundles. kate, ark, spectacle were not added to desktop-kde (yet?), but you they can now be installed with swupd bundle-add (i.e., no need to install the entire desktop-kde-apps bundle).

**Comment by thiagomacieira** [comment]
> I've added both the dev bundle will take another workday since those get processed daily

I don't see a `kde-frameworks5-dev` bundle. What is required to create one?

**Comment by rbharathkumar** [comment]
> Hi @rbharathkumar , The flatpak update dealing with TLS issue has been published (I did not post update about that back then, I apologize.). I'll double check the glib error you posted (I does not happen on my system, but I'll check on a fresh env).
> 
> Ongoing changes in bundles. kate, ark, spectacle were not added to desktop-kde (yet?), but you they can now be installed with swupd bundle-add (i.e., no need to install the entire desktop-kde-apps bundle).

Absolutely no need to apologize. You did inform me about the TLS update on flatpak and it was after that I had created a new VM to test out the new image and reported the GLIB error which is still there...Please let me know once it is fixed and as mentioned earlier we can test Discover's capability to handle additional repositories..

It is a wonderful decision to include Kate and Ark as separate bundles. I had not realized that it was available. I would expect Spectacle also to be available, but I anyways see gnome-screenshot available for now...so I hope it would work out until Spectacle is available. I will test it out and let you know for any  issues...I was really happy to see a separate desktop-lxqt available as well...Great work guys.. 👍 

P.S. Now all that is remaining is a proper desktop_xfce with pre-configured lightdm settings, variety (wallpaper changer)

**Comment by anselmolsm** [comment]
flatpak issue mentioned in https://github.com/clearlinux/distribution/issues/381#issuecomment-462007913 is related to #619

**Comment by watson96** [comment]
plasma-browser-integration also doesn't seems to be working here with the chrome browser(version 80.0.3987.87). I have the chrome extension installed.

**Comment by watson96** [comment]
Why doesn't plasma discover not show swupd bundles from CL's repo? I recently switched from GNOME desktop to KDE and found this inconsistency. Please guide me as to what might have gone wrong here.

**Comment by anselmolsm** [comment]
> Hi @anselmolsm I thought since this is a plasma-destop related issue I should just add this here. plasma-browser-integration also doesn't seems to be working here with the chrome browser(version 80.0.3987.87). I have the chrome extension installed.

Hi @watson96 , as of now Chrome is not integrated in Clear Linux.

**Comment by anselmolsm** [comment]
> Why doesn't plasma discover show swupd bundles from CL's repo? I recently switched from GNOME desktop to KDE and found this inconsistency. Please guide me as to what might have gone wrong here.
> Also, my print screen button has been doing nothing at all even though I have the spectacle app installed and I have all the keyboard shortcuts configured.

Plasma Discover needs a plugin to deal with CL's bundles and swupd. This plugin is not implemented.

Print screen: Can you take screenshots using spectacle UI directly? Is that a wayland session? (can you please create a separate issue for this?)

**Comment by watson96** [comment]
> Plasma Discover needs a plugin to deal with CL's bundles and swupd. This plugin is not implemented.

ookie dokie

> Print screen: Can you take screenshots using spectacle UI directly? Is that a wayland session? (can you please create a separate issue for this?)

It's the xorg session I'm using. I'll create a separate issue for this if that's what is appropriate.

Also another issue (according to me) regarding plasma desktop is kde desktop apps bundle containing 2 of some apps. For eg. there are two versions of the marble virtual globe app: one from kde and one from Qt. I am also not able to remove those apps via the swupd utility. Thanks for the response.

**Comment by anselmolsm** [comment]
> Also another issue (according to me) regarding plasma desktop is kde desktop apps bundle containing 2 of some apps. For eg. there are two versions of the marble virtual globe app: one from kde and one from Qt. I am also not able to remove those apps via the swupd utility. Thanks for the response.

Hm, that needs to be revisited, will track as #1787 

I'm closing this one as it's getting harder to keep track of all issues reported in comments. Please report new problems in separate issues.
